geos: handle POINT EMPTY output from GEOS

otan opened the pull request:
There is technically no valid way to encode POINT EMPTY in WKB, so GEOS
freaks out and returns an error. We can mitigate this by making it
return the GeoPackage spec compatible version by posting the raw NaNs
instead.

Upgrade twpayne/go-geom to be able to interpret empty points in the way
we expect them to be returned from PostGIS.

Resolves #49209.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50504)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 11 of 11 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50504#-:-MAWKpfZ0bPkqvhjQJu0:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geos/geos.cc, line 346 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50504#-MAWJ2Ct1E3r797ZuRbz:-MAWJ2Ct1E3r797ZuRc-:bupqajg) ([raw file](https://github.com/cockroachdb/cockroach/blob/15531ee8ab772bd204f159d359465050551ceb24/pkg/geo/geos/geos.cc#L346)):*
> ```cpp
> void CR_GEOS_writeGeomToEWKB(CR_GEOS* lib, CR_GEOS_Handle handle, CR_GEOS_Geometry geom,
>                              CR_GEOS_String* ewkb, int srid) {
>   // Empty points error in GEOS EWKB encoding. As such, we have to decode ourselves.
> ```

Did you mean encode instead of decode?
___
*[pkg/sql/logictest/testdata/logic_test/geospatial, line 414 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50504#-MAWKd-84FsIz0Q-_noT:-MAWKd-84FsIz0Q-_noU:bza272b) ([raw file](https://github.com/cockroachdb/cockroach/blob/15531ee8ab772bd204f159d359465050551ceb24/pkg/sql/logictest/testdata/logic_test/geospatial#L414)):*
> ```
>   ('Empty Point', 'POINT EMPTY'),
>   ('Empty GeometryCollection', 'GEOMETRYCOLLECTION EMPTY')
>   -- ('Partially Empty GeometryCollection', 'GEOMETRYCOLLECTION ( LINESTRING EMPTY, POINT (0.0 0.0) )') -- some of these cases crashes GEOS.
> ```

s/crashes/crash/


<!-- Sent from Reviewable.io -->

otan committed:
geos: handle POINT EMPTY output from GEOS

There is technically no valid way to encode POINT EMPTY in WKB, so GEOS
freaks out and returns an error. We can mitigate this by making it
return the GeoPackage spec compatible version by posting the raw NaNs
instead.

Upgrade twpayne/go-geom to be able to interpret empty points in the way
we expect them to be returned from PostGIS.

Release note: None

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50504)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[pkg/geo/geos/geos.cc, line 346 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50504#-MAWJ2Ct1E3r797ZuRbz:-MAWgG9v5JDGUmCKQM82:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/15531ee8ab772bd204f159d359465050551ceb24/pkg/geo/geos/geos.cc#L346)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

Did you mean encode instead of decode?
</blockquote></details>

Done.
___
*[pkg/sql/logictest/testdata/logic_test/geospatial, line 414 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50504#-MAWKd-84FsIz0Q-_noT:-MAWgH-O0m1bscIWGVjA:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/15531ee8ab772bd204f159d359465050551ceb24/pkg/sql/logictest/testdata/logic_test/geospatial#L414)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

s/crashes/crash/
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola
